odoo/models.py

```diff
@@ -1674,22 +1667,20 @@ def _read_group_fill_results(self, domain, groupby, remaining_groupbys,
                                  read_group_result, read_group_order=None):
         """Helper method for filling in empty groups for all possible values of
            the field being grouped by"""
+        field = self._fields[groupby]
+        if not field.group_full:
+            return read_group_result
 
-        # self._group_by_full should map groupable fields to a method that returns
-        # a list of all aggregated values that we want to display for this field,
-        # in the form of a m2o-like pair (key,label).
-        # This is useful to implement kanban views for instance, where all columns
-        # should be displayed even if they don't contain any record.
+        # field.group_full is the name of a method that returns a list of all
+        # aggregated values that we want to display for this field, in the form
+        # of a m2o-like pair (key,label).
+        # This is useful to implement kanban views for instance, where all
+        # columns should be displayed even if they don't contain any record.
 
         # Grab the list of all groups that should be displayed, including all present groups
-        present_group_ids = [x[groupby][0] for x in read_group_result if x[groupby]]
-        all_groups, folded = self._group_by_full[groupby](
-            # Beware: present_group_ids do not belong to model self!
-            self.browse(present_group_ids),
-            domain,
-            read_group_order=read_group_order,
-            access_rights_uid=odoo.SUPERUSER_ID,
-        )
+        group_ids = [x[groupby][0] for x in read_group_result if x[groupby]]
+        groups = self.env[field.comodel_name].browse(group_ids)
+        all_groups, folded = getattr(self, field.group_full)(groups, domain, read_group_order)
```

Comment:
Maybe pass in the third parameter as kw (and `order` or `orderby`) to match normal search protocol?

And couldn't the folding state of a given group be returned alongside the group itself?